1_fetch/src/get_site_data.R

```diff
@@ -1,6 +1,7 @@
 # download data for each site
 # packages needed: tidyverse, dataRetrieval
-get_site_data <- function(site_info, state, parameter) {
+get_site_data <- function(site_info, state, parameter) retry (
```

Comment:
Oh wow - I've never seen a function call written into a function definition before. Does this work as expected?!

The idea here is to decouple the retry from the actual pull code so that it is easily parameterizable by a user of the pipeline, so we were thinking this would be implemented in one of two ways - with `get_site_data()` wrapped by `retry()` in the target definition, like so:
```r
 tar_target(nwis_data,
   retry(get_site_data(nwis_inventory, nwis_inventory$state_cd, parameter),
        when = 'Ugh, the internet data transfer failed! Try again.',
        max_tries = 30),
   pattern = map(nwis_inventory)
 ),
```

Or by creating a wrapper function:
```r
get_nwis_data_retry <- function(site_info, state, parameter) {
  retry(
    get_site_data(nwis_inventory, nwis_inventory$state_cd, parameter),
    when = 'Ugh, the internet data transfer failed! Try again.',
    max_tries = 30
  )
}
```

Reply:
The way I wrote it works because `retry()` can take an entire code block and not just a function call for its `expr` argument. I was able to see it working because certain states' site data downloads would fail and be re-attempted several times.

But, the way you suggested is far clearer and better! I'm switching to it right away.

Thanks!

Reply:
Oh interesting -- that's something I've not encountered before!